app/src/main/java/com/google/maps/android/ktx/demo/MainActivity.kt

```diff
@@ -0,0 +1,236 @@
+        // Shared object managers - used to support multiple layer types on the map simultaneously
+        val markerManager = MarkerManager(map)
+        val groundOverlayManager = GroundOverlayManager(map)
+        val polygonManager = PolygonManager(map)
+        val polylineManager = PolylineManager(map)
+
+        addClusters(map, markerManager)
+        addGeoJson(map, markerManager, polylineManager, polygonManager, groundOverlayManager)
+        addKml(map, markerManager, polylineManager, polygonManager, groundOverlayManager)
+        addMarker(markerManager)
+    }
+
+    private fun addClusters(map: GoogleMap, markerManager: MarkerManager) {
+        val clusterManager = ClusterManager<MyItem>(this, map, markerManager)
+        map.setOnCameraIdleListener(clusterManager)
+
+        try {
+            val items = MyItemReader().read(resources.openRawResource(R.raw.radar_search))
+            clusterManager.addItems(items)
+        } catch (e: JSONException) {
+            Toast.makeText(this, "Problem reading list of markers.", Toast.LENGTH_LONG).show()
+            e.printStackTrace()
+        }
+    }
+
+    private fun addGeoJson(
+        map: GoogleMap,
+        markerManager: MarkerManager,
+        polylineManager: PolylineManager,
+        polygonManager: PolygonManager,
+        groundOverlayManager: GroundOverlayManager
+    ) {
+        // GeoJSON polyline
+        val geoJsonLineLayer = geoJsonLayer(
+            map = map,
+            resourceId = R.raw.south_london_line_geojson,
+            context = this,
+            markerManager = markerManager,
+            polygonManager = polygonManager,
+            polylineManager = polylineManager,
+            groundOverlayManager = groundOverlayManager
+        )
+        // Make the line red
+        val geoJsonLineStringStyle = GeoJsonLineStringStyle()
+        geoJsonLineStringStyle.color = Color.RED
+        for (f in geoJsonLineLayer.features) {
+            f.lineStringStyle = geoJsonLineStringStyle
+        }
+        geoJsonLineLayer.addLayerToMap()
+        geoJsonLineLayer.setOnFeatureClickListener { feature ->
+            Toast.makeText(
+                this,
+                "GeoJSON polyline clicked: " + feature.getProperty("title"),
+                Toast.LENGTH_SHORT
+            ).show()
+        }
+
+        // GeoJSON polygon
+        val geoJsonPolygonLayer = geoJsonLayer(
+            map = map,
+            resourceId = R.raw.south_london_square_geojson,
+            context = this,
+            markerManager = markerManager,
+            polygonManager = polygonManager,
+            polylineManager = polylineManager,
+            groundOverlayManager = groundOverlayManager
+        )
+        // Fill it with red
+        val geoJsonPolygonStyle = GeoJsonPolygonStyle()
+        geoJsonPolygonStyle.fillColor = Color.RED
+        for (f in geoJsonPolygonLayer.features) {
+            f.polygonStyle = geoJsonPolygonStyle
+        }
```

Comment:
Similarly, this can be written as:

```kotlin
geoJsonPolygonLayer.features.forEach {
    it.polygonStyle = GeoJsonPolygonStyle().apply {
        fillColor = Color.RED
    }
}
```

Reply:
Refactored in https://github.com/googlemaps/android-maps-ktx/pull/29/commits/89b22e8ddd52802b8cc6b102c66d11f3a540c589.